gluonnlp/model/cache.py

```diff
@@ -0,0 +1,134 @@
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Cache model."""
+__all__ = ['CacheCell']
+
+from mxnet import nd
+from mxnet.gluon import Block
+
+class CacheCell(Block):
+    """Cache language model.
+
+    Parameters
+    ----------
+    pretrained_lm_model : StandardRNN or AWDRNN
+        The type of RNN to use. Options are 'StandardRNN', 'StandardRNN'.
+    vocab_size : int
+        Size of the input vocabulary.
+    window : int
+        Size of cache window
+    theta : float
+        Mix between uniform distribution and cache softmax distribution over previous words
+    lambdas : float
+        Linear mix between only cache (1) and only vocab (0) distribution
```

Comment:
lambda?

Reply:
lambda is a specific symbol in python. This is intentional